pyriemann/utils/mean.py

```diff
@@ -368,15 +370,18 @@ def mean_power(covmats, p, *, sample_weight=None, zeta=10e-10):
     else:
         X = sqrtm(G)
 
-    test = 10 * zeta
-    while test > zeta:
+    crit = 10 * zeta
+    for k in range(maxiter):
         H = np.einsum(
             'a,abc->bc',
             sample_weight,
             powm(X @ powm(covmats, np.sign(p)) @ X.T, np.abs(p))
         )
         X = powm(H, -phi) @ X
-        test = np.linalg.norm(H - np.eye(n_channels)) / np.sqrt(n_channels)
+
+        crit = np.linalg.norm(H - np.eye(n_channels)) / np.sqrt(n_channels)
+        if crit <= zeta:
+            break
```

Comment:
usually I would advise to use an else block after the for
https://book.pythontips.com/en/latest/for_-_else.html#else-clause
and raise a warning if convergence was no reached.